posts/2021/04/rethinking-jupyter-documentation.md

```diff
@@ -0,0 +1,180 @@
+<!-- 
+.. title: Rethinking Jupyter Interactive Documentation
+.. slug: rethinking-jupyter-documentation
+.. date: 2021-03-30 11:59 UTC
+.. tags: Python, Open-Source, documentation
+.. category: 
+.. link: 
+.. description: 
+.. type: markdown
+-->
+
+Jupyter Notebook first release was 8 years ago – under the IPython Notebook
+name at the time. Even if notebooks were not invented by Jupyter; they were
+definitely democratized by it. Being Web powered allowed development of many
+changes in the Datascience world. Objects now often expose rich representation; from
+Pandas dataframes with as html tables, to more recent [Scikit-learn model](https://github.com/scikit-learn/scikit-learn/pull/14180).
+
+Today I want to look into a topic that has not evolved much since, and I believe
+could use an upgrade. Accessing interactive Documentation when in a Jupyter
+session, and what it could become.
+
+<!-- TEASER_END -->
+
+# The current limitation for users
+
+The current documentation of IPython and Jupyter come in a few forms, but mostly
+have the same limitation. 
+The typical way to reach for help is to use the `?` operator. Depending on
+the frontend you are using it will bring a pager, or a panel that will display
+some information about the current object. Here is the documentation for
+``numpy.linspace``
+
+
+<img alt="numpy.linspace help in IPython" src="/images/2021/04/numpy-linspace-current.png" width="600px" >
```

Comment:
This and the next screenshot are both quite large (in kilobytes), and I think this one may anyway look better as typeset code rather than a screenshot.

Reply:
I've merged then in a single image, where I show numpy.org and jupyterlab's help for a more impactful effect. 

I'll scale down the images last, as it's easy to scaledown an image, hard to scale up.